### PR TITLE
ENG-7925 - Prevent Multiple RegisteredUserIDs from being set

### PR DESCRIPTION
**Pull request by kevinsites**

### CODE REVIEW

**Review by cometman — DISMISSED** [review]

File: NeuroID/NeuroIDClass/Extensions/NIDUser.swift [hunk 1]
```diff
@@ -124,6 +124,12 @@ public extension NeuroID {
     }
 
     static func setRegisteredUserID(_ registeredUserID: String) -> Bool {
+        if !NeuroID.registeredUserID.isEmpty, registeredUserID != NeuroID.registeredUserID {
+            NeuroID.saveEventToLocalDataStore(NIDEvent(level: "warn", m: "Multiple Registered User Id Attempts"))
+            NIDLog.e("Multiple Registered UserID Attempt: Only 1 Registered UserID can be set per session")
+            return false
+        }
+
```
Comment (quacke-nid) [inline NeuroID/NeuroIDClass/Extensions/NIDUser.swift:132]: is `NeuroID.registeredUserID` reset to the empty string where appropriate (new session?)?

**Review by jeff-niemann — DISMISSED** [review]
🚀

**Review by quacke-nid — DISMISSED** [review]
Approved, but please check my comment.

File: NeuroID/NeuroIDClass/Extensions/NIDUser.swift (hunk 1 above)
Comment (jeff-niemann) [reply NeuroID/NeuroIDClass/Extensions/NIDUser.swift:132]: ref: https://github.com/Neuro-ID/trackjs/blob/main/src/collection/collector-api.js#L951

Comment (kevinsites) [reply NeuroID/NeuroIDClass/Extensions/NIDUser.swift:132]: Correct, when stop/stopSession is called we clear all session variables (userID, registeredUserID, and linkedSiteID)

**Review by kishisaka-nid — DISMISSED** [review]

**Review by cometman — APPROVED** [review]

**Pull request merged**